### PR TITLE
update datadog_lambda_forwarder ref for darwin_arm64

### PR DESCRIPTION
**Pull request by kevcube**
## what
* update datadog-lambda-forwarder module for darwin_arm64

## why
* run on Darwin_arm64 hardware

### CODE REVIEW

File: modules/s3-bucket/versions.tf [hunk 1]
```diff
@@ -8,7 +8,7 @@ terraform {
     }
     template = {
       source  = "cloudposse/template"
-      version = "2.2.0"
+      version = ">= 2.2.0"
```
Comment (kevcube) [inline modules/s3-bucket/versions.tf:11]: idk, bats was failing over version pinning.

**Comment by Gowiem** [comment]
@Nuru @milldr -- friendly ping here. We're seeing the `bats` workflow run bats tests for ALL of the components and not just those affected by this PR, which causes this PR to fail. [See this check for what I'm talking about](https://github.com/cloudposse/terraform-aws-components/actions/runs/4631351970/jobs/8194107122?pr=626). 

Ya'll are the two main contributors to https://github.com/cloudposse/terraform-aws-components/blob/master/.github/workflows/bats.yml, so wondering if you've seen that before and if that is a problem we need to deal with as part of this PR or if we should just ignore?

**Comment by milldr** [comment]
@Gowiem Bats has a newer requirement for version pinning that is missing from some less recently updated components. Since you've updated `s3-bucket`, you'll need to resolve the checks for that component. Specifically this needs to be updated for `s3-bucket` `versions.tf`
```
    aws = {
      source  = "hashicorp/aws"
      version = ">= 4.0"
    }
```

**Comment by milldr** [comment]
Pushing a PR now to resolve bats across all components. See #627

**Comment by milldr** [comment]
#627 is merged now. Pls updated this branch

**Comment by kevcube** [comment]
done @milldr

**Comment by milldr** [comment]
It appears that since this PR is from a fork, `bats` is running for all components, so we have to have all resolved before we can merge this PR. Looks like we missed a few provider version pins on a few other components, so I'll push 1 more PR. Hold tight

**Comment by Gowiem** [comment]
Thanks for the help @milldr 🥇

**Comment by milldr** [comment]
1 more PR is needed 🙁

**Review by osterman — APPROVED** [review]

**Comment by milldr** [comment]
we can merge this with the tests passing for now (although we'll have those resolved soon). thanks for the contribution!

**Comment by kevcube** [comment]
thanks a bunch @milldr

**Pull request merged**